### DISCUSSION TITLE
Pico flash results in hard hang every time

### DISCUSSION THREAD

**Discussion by Erik84750** [earlephilhower/arduino-pico#2586] [General]
Having installed the Additional Boards Manager url, and installed the Pico boards, and having performed the first-time BOOTSEL procedure then subsequent flashing fail due to hard hang of the Pico board after USB connection and at flash start.
Only using the BOOTSEL procedure every time before flashing works.

How to reset the Pico back to factory status, or else: how to solve this hard hang issue without having to use the BOOTSEL button procedure every time??

**Comment by earlephilhower** [comment]
Some basic sanity tests need to be performed.  First of all, is this a real Raspberry Pi branded Pico board (what model) or a clone?  If a clone, did you select the clone from the boards menu?  When you upload the `blink` example, does it blink?  Does the USB serial port appear on your PC?

**Comment by Erik84750** [comment]
Hi thanks for your very fast reply!
No, it is an original Pico W. When flashing Blink.ino (with the BOOTSEL depressed on USB connection) the built-in led blinks.
Flashing occurs when selecting "UF2 Board" under "UF2 Devices", no serial port.
When removing the USB cable and replugging without the BOOTSEL "procedure" then a Serial Com's port appears; in the Arduino IDE this port is identified as "COM3 (Raspberry Pi Pico W)".

But just after my OP and before this post I had been doing some research and found how to flash an uf2 file called "flash_nuke.uf2".
Subsequently clearing the complete USB cached drivers and data related to RPi Pico according to this Arduino forum post:
https://forum.arduino.cc/t/raspberry-pi-pico-cant-load-any-sketch/1008640/9

At this moment I can plug in the Pico (without BOOTSEL pressed) and flash the blink.ino without issues.

I need to investigate further (tomorrow) on my other two Pico's to verify if the solution is solid, and report back.

**Comment by Erik84750** [comment]
Confirmed solved on three Pico W boards.

In conclusion, when hard hangs are experienced every time even after using the BOOTSEL button procedure, I found the above solution to be solid.
